app/controllers/api_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class ApiController < ApplicationController
+  before_action :verify_token
+
+  def index
+    render json: { status: 'ok' }
+  end
+
+private
+
+  def verify_token
+    access_token = parse_access_token(request.headers['AUTHORIZATION'])
+
+    if valid_token?(access_token)
+      true
+    else
+      render json: { status: 'error' }
+    end
+  end
+
+  def parse_access_token(auth_header)
+    return nil if auth_header.nil?
+
+    return nil unless auth_header.starts_with?('Bearer')
+
+    auth_header.split.last
+  end
+
+  def valid_token?(access_token)
+    return false if Nomis::Oauth::Token.new(access_token: access_token).expired?
+
+    if Nomis::Oauth::Token.new(access_token: access_token).valid_token?
+      true
+    else
+      false
+    end
```

Comment:
So I don't think you need this block.  The if statement returns a boolean, and then you're going to return the same boolean.  You should be able to replace it all with

```suggestion
Nomis::Oauth::Token.new(access_token: access_token).valid_token?
```
Having said that, I'd check the valid_token? first, and just put the expired check inside valid_token?